### PR TITLE
[Platform API] Get and wait for Jobs & Reload App Service Configuration

### PR DESCRIPTION
**Pull request by quaark**
Add interface for reloading app services configuration (which returns a job response) and waiting on the job to complete

### CODE REVIEW

**Review by Hedingber — APPROVED** [review]

File: pkg/controlplane/http/session.go [hunk 1]
```diff
@@ -245,6 +246,99 @@ func (s *session) GetRunningUserAttributesSync(getRunningUserAttributesInput *v3
...
+		&reloadAppServicesConfigInput.ControlPlaneInput,
+		map[string]string{},
+		&reloadAppServicesConfigOutput.ControlPlaneOutput,
+		&reloadAppServicesConfigOutput.JobAttributes)
+
+	if err != nil {
+		return "", err
+	}
+
+	return reloadAppServicesConfigOutput.ID, nil
```
Comment (omesser) [inline pkg/controlplane/http/session.go:281]: Look like a mistake, it's an attribute called job_id

File: pkg/controlplane/http/session.go [hunk 2]
```diff
@@ -245,6 +246,99 @@ func (s *session) GetRunningUserAttributesSync(getRunningUserAttributesInput *v3
...
+
+	for time.Now().Before(deadline) {
+		getJobsOutput, err := s.GetJobs(&getJobsInput)
+		if err != nil {
+			return errors.Wrap(err, "Failed getting job")
+		}
+
+		switch getJobsOutput.State {
+		case v3ioc.JobStateCompleted:
+			return nil
```
Comment (omesser) [inline pkg/controlplane/http/session.go:303]: log?

File: pkg/controlplane/http/session.go [hunk 3]
```diff
@@ -245,6 +246,99 @@ func (s *session) GetRunningUserAttributesSync(getRunningUserAttributesInput *v3
...
+		case v3ioc.JobStateFailed:
+			return errors.New("Job has failed")
+		case v3ioc.JobStateCanceled:
+			return errors.New("Job was canceled")
+		default:
+			s.logger.DebugWithCtx(ctx, "Job in progress",
+				"jobId", jobId,
+				"retryInterval", retryInterval,
+				"timeout", timeout)
+			time.Sleep(retryInterval)
```
Comment (omesser) [inline pkg/controlplane/http/session.go:313]: //TODO - backoff

File: pkg/controlplane/types.go [hunk 4]
```diff
@@ -151,3 +163,25 @@ type GetRunningUserAttributesOutput struct {
...
+
+// ReloadAppServicesConfigInput specifies how to reload the app services configuration
+type ReloadAppServicesConfigInput struct {
+	ControlPlaneInput
+}
+
+// ReloadAppServicesConfigOutput specifies holds the response from reload app services config
+type ReloadAppServicesConfigOutput struct {
+	ControlPlaneOutput
+	JobAttributes
```
Comment (omesser) [inline pkg/controlplane/types.go:186]: bug?

**Review by omesser — CHANGES_REQUESTED** [review]

File: pkg/controlplane/http/session.go (hunk 1 above)
Comment (quaark) [reply pkg/controlplane/http/session.go:281]: It isn't a mistake, what's returned is:
```
{
    "id": "<id>",
    "type": "job",
}

File: pkg/controlplane/types.go (hunk 4 above)
Comment (quaark) [reply pkg/controlplane/types.go:186]: Nope, that is the schema that is returned

File: pkg/controlplane/http/session.go (hunk 1 above)
Comment (omesser) [reply pkg/controlplane/http/session.go:281]: my, bad, cluster reloads still acts the "old" way and only it returns custom schema

File: pkg/controlplane/types.go (hunk 4 above)
Comment (omesser) [reply pkg/controlplane/types.go:186]: Please rename to match - ReloadAppServicesConfigOutput -> ReloadAppServicesConfigJobOutput

**Review by omesser — APPROVED** [review]

**Pull request merged**